rest/route/project_test.go

```diff
@@ -776,3 +776,45 @@ func TestDeleteProject(t *testing.T) {
 	resp = pdh.Run(ctx)
 	assert.Equal(t, http.StatusBadRequest, resp.Status())
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// Tests for PUT /rest/v2/projects/variables/rotate
+
+type ProjectPutRotateSuite struct {
+	sc *data.MockConnector
+	rm gimlet.RouteHandler
+
+	suite.Suite
+}
+
+func TestProjectPutRotateSuite(t *testing.T) {
+	suite.Run(t, new(ProjectPutRotateSuite))
+}
+
+func (s *ProjectPutRotateSuite) SetupTest() {
+	s.NoError(db.ClearCollections(serviceModel.RepoRefCollection, user.Collection))
+	s.sc = getMockProjectsConnector()
+
+	settings, err := evergreen.GetConfig()
+	s.NoError(err)
+	s.rm = makeProjectVarsPut(s.sc, settings).(*projectVarsPutHandler)
+}
+
+func (s *ProjectPutRotateSuite) TestRotateProjectVars() {
+	ctx := context.Background()
+	json := []byte(
+		`{
+				"to_replace": "yellow",
+				"replacement": "brown"
+		}`)
+
+	req, _ := http.NewRequest("PUT", "http://example.com/api/rest/v2/projects/variables/rotate", bytes.NewBuffer(json))
+	err := s.rm.Parse(ctx, req)
+	s.NoError(err)
+	resp := s.rm.Run(ctx)
+	s.NotNil(resp)
+	s.NotNil(resp.Data())
+	s.Equal(resp.Status(), http.StatusOK)
+	s.Equal("brown", s.sc.CachedVars[0].Vars["banana"])
```

Comment:
Can we verify that when "dry_run" is passed in the value isn't modified?

Reply:
done